models/hashtag.go

```diff
@@ -20,6 +21,42 @@ type Hashtag struct {
 	UpdatedUnix  int64
 }
 
+// Gets a summary of the hashtags for repositories with the specified prefix.
+// The `engine` parameter is for unit testing.
+// If `engine` is nil, then the `x` engine from models.go is used.
+func GetHashtagSummary(engine *xorm.Engine, repoPrefix string) ([]map[string]string, error) {
```

Comment:
Would it be possible to mock out the SQL engine some other way? It would be nice if we didn't have to pass in a `nil` argument every time this function is called in a non-test setting, particularly because all user of this function will have to remember that it takes an extra argument.